Update docs (#153)
deepqmc/deepqmc: doc/cli.rst

Hyperparameters
---------------

In the following the most relevant settings for running experiments with DeepQMC are discussed.

Task
____

DeepQMC provides the above mentioned conigurations for the ``train``, ``evaluate`` and ``restart`` task. In order to override default hyperparameters of the experimental setup, such as the ``sample_size`` or the number of training ``steps`` or ``pretrain_steps``, hydra provides a simple syntax::

>         $ deepqmc task=train task.sample_size=2048 task.steps=50_000 task.pretrain_steps

Reviewer: I don't think the `pretrain_steps` argument is correct.

1. It's missing a value
2. I think it needs the `+` sign in front.

Author: Thanks, yes something got messed up there.

Reviewer 2: I've fixed this argument in my last commit. I used `+task.pretrain_steps=1000`

The working directory for logging and checkpointing is is defined through::

        $ deepqmc hydra.run.dir=workdir

> Note that the working directory of an ``evaluate`` and ``restart`` task cannot match the ``restdir``.

Reviewer: ```suggestion
Note that the working directory of an ``evaluate`` and ``restart`` task cannot match the value of their ``restdir`` option (which specifies the working directory of the previous job that we want to evaluate or restart).
```

Hamiltonian
___________

> DeepQMC aims at solving the molecular Hamiltonian. Molecules can be selected from a range of predefined configurations located in ``.../deepqmc/src/deepqmc/conf/ansatz``::

Reviewer: ```suggestion
DeepQMC aims at solving the molecular Hamiltonian. Molecules can be selected from a range of predefined configurations located in ``.../deepqmc/src/deepqmc/conf/hamil/mol``::
```

        $ deepqmc hamil/mol=LiH

The predefined configurations can be extended with custom molecules. Alternatively molecules can be specified on the command line::

        $ deepqmc hamil.mol.coords=[[0,0,0],[0.742,0,0]] hamil.mol.charges=[1,1] hamil.mol.charge=0 hamil.mol.spin=0 hamil.mol.unit=angstrom

Furthermore, DeepQMC implements the option to use pseudopotentials, which can be used via::

        $ deepqmc hamil.mol.coords=[[0,0,0]] hamil.mol.charges=[20] hamil.mol.charge=0 hamil.mol.spin=0 hamil.mol.unit=angstrom +hamil.mol.pp_type='ccECP'

Sampling
________

> Different sampler configurations can be found in ``.../deepqmc/src/deepqmc/conf/ansatz``. A typical usecase would be to pick as sampler form these configurations and , if required, change some argument from the command line::

Reviewer: ```suggestion
Different sampler configurations can be found in ``.../deepqmc/src/deepqmc/conf/task/sampler``. A typical usecase would be to pick as sampler form these configurations and, if required, change some argument from the command line::
```

        $deepqmc task/sampler=decorr_langevin task.sampler.0.length=30

Optimization
____________

For the optimization either `KFAC <https://kfac-jax.readthedocs.io/en/latest/>`_ or optimizers from `optax <https://optax.readthedocs.io/en/latest/>`_ may be used. While the use of `KFAC <https://kfac-jax.readthedocs.io/en/latest/>`_ is highly recommended due to the significantly imporved convergence, at times it can be useful to run with other optimizers such as `AdamW <https://optax.readthedocs.io/en/latest/api.html#adamw>`_::

        $ deepqmc task/opt=adamw

Ansatz
______

The hyperparameters of the training and the wave function ansatz are specified through hydra config files. Predefined ansatzes can be found in ``.../deepqmc/src/deepqmc/conf/ansatz`` and selected via::

    $ deepqmc ansatz=ferminet

The hyperparameters of such a predifined ansatz can also be overwritten at the command line::

>     $ deepqmc ansatz.omni_factory.gnn_factory.n_interactions=2

Reviewer: ```suggestion
    $ deepqmc ansatz=ferminet ansatz.omni_factory.gnn_factory.n_interactions=2
```
It might be clearer if we continue with the previous example.

Author: My incentive was to reduce the complexity of these commands, but I don't mind changing it back if you prefer that.
